server/api/initial_data.py

```diff
@@ -577,7 +577,7 @@ def _migrate_artifacts_batch(
             link_artifact_ids.append(artifact.id)
             continue
 
-        artifact_metadata = artifact_dict.get("metadata", None)
+        artifact_metadata = artifact_dict.get("metadata", {})
```

Comment:
Just to be extra safe I think you should do
```
artifact_dict.get("metadata") or {}
```
because `{"metadata": None}.get("metadata", {})` returns None